src/sage/rings/polynomial/multi_polynomial_ring.py

```diff
@@ -891,11 +888,34 @@ def addwithcarry(tempvector, maxvector, pos):
 
         while tempvector != maxvector:
             tempvector = addwithcarry(list(tempvector), maxvector, pos)
-            M.append(R(PolyDict({ETuple(tempvector): one},
-                                force_int_exponents=False,
-                                force_etuples=False)))
+            M.append(R(PolyDict({ETuple(tempvector): one})))
         return M
 
+    def sum(self, terms):
+        r"""
+        Return a sum of elements of this multipolynomial ring.
+
+        This is method is much faster than the Python builtin sum.
+
+        EXAMPLES::
+
+            sage: R = QQ['x']
+            sage: S = R['y, z']
+            sage: x = R.gen()
+            sage: y, z = S.gens()
+            sage: S.sum([x*y, 2*x^2*z - 2*x*y, 1 + y + z])
+            (-x + 1)*y + (2*x^2 + 1)*z + 1
+
+        Comparison with builtin sum::
+
+            sage: sum([x*y, 2*x^2*z - 2*x*y, 1 + y + z])
+            (-x + 1)*y + (2*x^2 + 1)*z + 1
+        """
+        ans = self(0)
```

Comment:
```suggestion
        ans = copy(self.zero())
```
or perhaps if there is a faster way to make a mutable copy. I didn't check which way is fastest.

Reply:
Why do you want to make a copy here?
```
sage: R = QQ['x']
sage: S = R['y, z']
sage: S(0) is S(0)
False
```

Reply:
Speed. Copying the zero object should be faster than going through the coercion framework and (generally) the input checks. (I also am slightly worried about the result not being the same as `S.zero()`; this would be a sensible optimization since elements are supposed to be immutable.)

Reply:
If elements are supposed immutable, how can you assume that `copy(S.zero())` is not the same object as `S.zero()`? The `copy` just calls `__copy__` on the object which can implement whathever. For example
```python
>>> from copy import copy
>>> a = 1
>>> copy(a) is a
True
```

Reply:
I perhaps should have said "or however one implements a copy". Usually there is a faster direct way to make a copy by directly using the class's constructor.

Reply:
Sure, I can do that.

Reply:
Hopefully better in https://github.com/sagemath/sage/pull/35174/commits/23d50f4d9a4b6da943dddee3c0ede1bb88f663fd.

Reply:
https://github.com/sagemath/sage/pull/35174/commits/dfd6609d86758031cad774edeb8b078cc2cdeba3 introduces a `check` argument.

Reply:
I think that is good. Thank you.